enterprise/app/workflows/github_app_import.tsx

```diff
@@ -0,0 +1,303 @@
+import { Octokit } from "@octokit/rest";
+import Long from "long";
+import { Check, ChevronsDown, ExternalLink } from "lucide-react";
+import React from "react";
+import alertService from "../../../app/alert/alert_service";
+import { User } from "../../../app/auth/user";
+import Banner from "../../../app/components/banner/banner";
+import FilledButton, { OutlinedButton } from "../../../app/components/button/button";
+import LinkButton from "../../../app/components/button/link_button";
+import { FilterInput } from "../../../app/components/filter_input/filter_input";
+import Select, { Option } from "../../../app/components/select/select";
+import Spinner from "../../../app/components/spinner/spinner";
+import errorService from "../../../app/errors/error_service";
+import router from "../../../app/router/router";
+import rpcService, { CancelablePromise } from "../../../app/service/rpc_service";
+import { normalizeRepoURL } from "../../../app/util/git";
+import { github } from "../../../proto/github_ts_proto";
+
+type GitHubAppImportProps = {
+  user: User;
+};
+
+type State = {
+  installationsResponse: { installations: github.AppInstallation[] } | null;
+  installationsLoading: boolean;
+
+  accessibleReposResponse: github.GetAccessibleReposResponse | null;
+  accessibleReposLoading: boolean;
+  repoListLimit: number;
+  searchQuery: string;
+
+  linkedReposLoading: boolean;
+  linkedReposResponse: github.IGetLinkedReposResponse | null;
+
+  selectedInstallation: github.AppInstallation | null;
+
+  linkRequest: github.ILinkRepoRequest | null;
+  linkLoading: boolean;
+};
+
+const SEARCH_DEBOUNCE_DURATION_MS = 250;
+const REPO_LIST_DEFAULT_LIMIT = 10;
+const REPO_LIST_SHOW_MORE_INCREMENT = 10;
+
+/**
+ * Displays a page that lets the user link a GitHub repository to BuildBuddy.
+ */
+export default class GitHubAppImport extends React.Component<GitHubAppImportProps, State> {
+  state: State = {
+    installationsLoading: false,
+    installationsResponse: null,
+
+    accessibleReposResponse: null,
+    accessibleReposLoading: false,
+
+    linkedReposLoading: false,
+    linkedReposResponse: null,
+    repoListLimit: REPO_LIST_DEFAULT_LIMIT,
+    searchQuery: "",
+
+    selectedInstallation: null,
+
+    linkLoading: false,
+    linkRequest: null,
+  };
+
+  private fetchInstallationsRPC?: CancelablePromise;
+  private linkedReposRPC?: CancelablePromise;
+  private searchRPC?: CancelablePromise;
+
+  componentDidMount() {
+    document.title = "Link GitHub repo | BuildBuddy";
+    this.fetch();
+  }
+
+  componentDidUpdate(prevProps: GitHubAppImportProps) {
+    if (this.props.user !== prevProps.user) this.fetch();
+  }
+
+  private octokit() {
+    return new Octokit({ auth: this.props.user.githubToken });
```

Comment:
What do we use Octokit for here?

Reply:
It's to fetch installations accessible to the user - I wasn't sure whether to show the installations accessible to the user, or the installations linked to the org, since the user might not have access to some of those installations. But thinking about it more, it's probably better to show only the installations linked to the org and just let the UI show an error state if the user doesn't have access to the installation. I can update the PR to use that approach instead

Reply:
Done (removed octokit in favor of fetching org-linked app installations)